packages/fetchai/skills/erc1155_deploy/handlers.py

```diff
@@ -237,8 +238,14 @@ def handle(self, message: Message) -> None:
         if tx_msg_response.tx_id == contract.Performative.CONTRACT_DEPLOY.value:
             tx_signed = tx_msg_response.signed_payload.get("tx_signed")
             tx_digest = self.context.ledger_apis.ethereum_api.send_signed_transaction(
-                is_waiting_for_confirmation=True, tx_signed=tx_signed
+                tx_signed=tx_signed
             )
+            # TODO; handle case when no tx_digest returned and remove loop
```

Comment:
Note, we cannot allow for blocking calls, hence the API has been changed. The skill still needs to be changed. and the while loop removed.